### PR TITLE
Some clean-up of tests to be more canonical

### PR DESCRIPTION
**Pull request by MichaelChirico**
Mostly being more consistent about using `testthat::test_path()`, and some style touch-ups

### CODE REVIEW

File: tests/testthat/test-lint_package.R [hunk 1]
```diff
@@ -16,17 +16,19 @@ test_that(
...
 
   expected_lines <- c(
     # from abc.R
     "abc = 123",
     # from jkl.R
-    "jkl = 456", "mno = 789"
+    "jkl = 456",
+    "mno = 789"
   )
   read_settings(NULL)
```
Comment (AshesITR) [inline tests/testthat/test-lint_package.R:28]: This actually tests the default settings. (Related to the bug in the `.lintr` PR?)
Better to `skip_if(!is.null(find_config(...)))` and let `lint_package()` notice that the default config is needed?

Requires copying the package to a tempdir so our `.lintr` isn't found.

Comment (MichaelChirico) [reply tests/testthat/test-lint_package.R:28]: sounds less trivial than this PR... follow-up? 😬

Comment (AshesITR) [reply tests/testthat/test-lint_package.R:28]: Should be straight-forward now with the new solution of using `lintr_test_config`. That file is unlikely to exist in parent directories by accident.

**Review by AshesITR — APPROVED** [review]

File: tests/testthat/test-lint_package.R [hunk 2]
```diff
@@ -13,20 +13,24 @@
 
 test_that(
   "`lint_package` does not depend on path to pkg - no excluded files", {
+
```
Comment (MichaelChirico) [inline tests/testthat/test-lint_package.R:16]: please check new lints

**Comment by MichaelChirico** [comment]
thanks for the clean-up... i was still trying to digest your suggestion :)

File: tests/testthat/test-lint_package.R (hunk 2 above)
Comment (AshesITR) [reply tests/testthat/test-lint_package.R]: sorry, GitHub web UI isn't exactly helpful there ^^

**Review by AshesITR — APPROVED** [review]

**Pull request merged**